ui/raidboss/data/05-shb/raid/e12s.js

```diff
@@ -400,43 +400,189 @@ export default {
       id: 'E12S Promise Chiseled Sculpture',
       netRegex: NetRegexes.headMarker({}),
       condition: (data, matches) => data.isDoorBoss && matches.target === data.me,
-      response: (data, matches, output) => {
-        // cactbot-builtin-response
-        output.responseOutputStrings = {
-          // This is sort of redundant, but if folks want to put "square" or something in the text,
-          // having these be separate would allow them to configure them separately.
-          square1: numberOutputStrings[1],
-          square2: numberOutputStrings[2],
-          square3: numberOutputStrings[3],
-          square4: numberOutputStrings[4],
-          triangle1: numberOutputStrings[1],
-          triangle2: numberOutputStrings[2],
-          triangle3: numberOutputStrings[3],
-          triangle4: numberOutputStrings[4],
-        };
-
+      run: (data, matches) => {
         const id = getHeadmarkerId(data, matches);
 
         // Statue laser mechanic.
         const firstLaserMarker = '0091';
         const lastLaserMarker = '0098';
         if (id >= firstLaserMarker && id <= lastLaserMarker) {
-          // We could arguably tell you which giant you're tethered to by finding their position?
-          // And then saying something like "North" but that's probably more confusing than helpful.
           // ids are sequential: #1 square, #2 square, #3 square, #4 square, #1 triangle etc
           const decOffset = parseInt(id, 16) - parseInt(firstLaserMarker, 16);
-          return {
-            alertText: [
-              output.square1(),
-              output.square2(),
-              output.square3(),
-              output.square4(),
-              output.triangle1(),
-              output.triangle2(),
-              output.triangle3(),
-              output.triangle4(),
-            ][decOffset],
-          };
+          data.statueTetherNumber = (decOffset % 4) + 1;
+        }
+      },
+    },
+    {
+      id: 'E12S Promise Chiseled Sculpture Collector',
+      netRegex: NetRegexes.addedCombatantFull({ npcNameId: '9818' }),
+      run: (data, matches) => {
+        // Collect both sculptures up front, so when we find the tether on the
+        // current player we can look up both of them immediately.
+        data.statueIds = data.statueIds || [];
+        data.statueIds.push(parseInt(matches.id, 16));
+      },
+    },
```

Comment:
This is the only thing that jumps out at me as iffy. I'm not sure if the log lines for this happen at fade in or during the fight, if it's the latter then the trigger will fail if it's started after zone-in but before pull.

Reply:
These happen mid-fight:
```
[20:07:59.272] 03:4000900B:Added new combatant Chiseled Sculpture.  Job: N/A Level: 80 Max HP: 148000 Max MP: 10000 Pos: (0,-75,75) (98180000012445).
[20:07:59.272] 03:4000900A:Added new combatant Chiseled Sculpture.  Job: N/A Level: 80 Max HP: 148000 Max MP: 10000 Pos: (0,-75,75) (98180000012445).
[20:08:04.397] 16:4000900B:Chiseled Sculpture:58B2:Classical Sculpture:etc
```

Reply:
Alright. I wasn't at my computer to verify this at the time.

Reply:
Thanks for the question.  There's been a lot of things added early and then moved later this tier, so it's good to check.  :)